plugins/volumeFilter.go

```diff
@@ -175,74 +175,95 @@ func (f *volumeFilter) Apply(ops []txnbuild.Operation, sellingOffers []hProtocol
-			keepSellingQuote = projectedSoldInQuoteUnits <= *lp.baseAssetCapInQuoteUnits
-			newAmountString := ""
-			if lp.mode == volumeFilterModeExact && !keepSellingQuote {
-				newAmount := (*lp.baseAssetCapInQuoteUnits - *dailyOTB.BaseAssetCapInQuoteUnits - *dailyTBBAccumulator.BaseAssetCapInQuoteUnits) / sellPrice
-				if newAmount > 0 {
-					newAmountBeingSold = newAmount
-					opToReturn.Amount = fmt.Sprintf("%.7f", newAmountBeingSold)
-					keepSellingQuote = true
-					newAmountString = ", newAmountString = " + opToReturn.Amount
-				}
-			}
-			log.Printf("volumeFilter: selling (quote units), price=%.8f amount=%.8f, keep = (projectedSoldInQuoteUnits) %.7f <= %.7f (config.BaseAssetCapInQuoteUnits): keepSellingQuote = %v%s", sellPrice, amountValueUnitsBeingSold, projectedSoldInQuoteUnits, *lp.baseAssetCapInQuoteUnits, keepSellingQuote, newAmountString)
-		} else {
-			keepSellingQuote = true
-		}
+	// if projected is under the cap, update the tbb and return the original op
+	projected := otb + tbb + offerAmount*capPrice
+	if projected <= cap {
+		*dailyTBBAccumulator.BaseAssetCapInBaseUnits += offerAmount
+		*dailyTBBAccumulator.BaseAssetCapInQuoteUnits += offerAmount * offerPrice
+		return op, nil
+	}
 
-		if keepSellingBase && keepSellingQuote {
-			// update the dailyTBB to include the additional amounts so they can be used in the calculation of the next operation
-			*dailyTBBAccumulator.BaseAssetCapInBaseUnits += newAmountBeingSold
-			*dailyTBBAccumulator.BaseAssetCapInQuoteUnits += (newAmountBeingSold * sellPrice)
-			return opToReturn, nil
-		}
-	} else {
-		// TODO buying side - we need to implement this to support buy side filters; extract common logic from the above sell side case
+	if lp.mode != volumeFilterModeExact {
+		return nil, nil
 	}
 
-	// we don't want to keep it so return the dropped command
-	return nil, nil
+	// if exact mode and with remaining capacity, update the op amount and return the op
+	// else, return nil
+	remainingCapacity := (cap - otb - tbb) / capPrice
+	if remainingCapacity <= 0 {
+		return nil, nil
+	}
+
+	offerAmount = remainingCapacity
+	op.Amount = fmt.Sprintf("%.7f", offerAmount)
+	*dailyTBBAccumulator.BaseAssetCapInBaseUnits += offerAmount
+	*dailyTBBAccumulator.BaseAssetCapInQuoteUnits += offerAmount * offerPrice
+	return op, nil
+}
+
+func offerSameTypeAsFilter(filter *VolumeFilterConfig, op *txnbuild.ManageSellOffer, baseAsset hProtocol.Asset, quoteAsset hProtocol.Asset) (bool, error) {
+	opIsSelling, e := utils.IsSelling(baseAsset, quoteAsset, op.Selling, op.Buying)
+	if e != nil {
+		return false, fmt.Errorf("error when running the isSelling check for offer '%+v': %s", *op, e)
+	}
+	filterIsSelling := filter.action == queries.DailyVolumeActionSell
+	return opIsSelling == filterIsSelling, nil
+}
+
+func extractFilterInputs(dailyOTB *VolumeFilterConfig, dailyTBB *VolumeFilterConfig, lp limitParameters) (float64, float64, float64, error) {
```

Comment:
whenever we return values of the same type we should comment what is being returned. the comment should be of the form /* name */ and should come *after* the type -- that's just how comments are best formatted in such a situation. you can leave out comments for non-repeated types such as error in this case